### PR TITLE
rework beaver builder original strings (WP-744)

### PR DESCRIPTION
**Pull request by vsolovei-smartling**
use snapshot of original strings instead of relying on current strings from source blog

### CODE REVIEW

File: inc/Smartling/ContentTypes/ExternalContentBeaverBuilder.php [hunk 1]
```diff
@@ -122,47 +122,61 @@ public function getRelatedContent(string $contentType, int $contentId): array
 
     public function setContentFields(array $original, array $translation, SubmissionEntity $submission): array
     {
-        $translation['meta'][self::META_FIELD_NAME] = $this->buildData(unserialize($original['meta'][self::META_FIELD_NAME]), $translation[$this->getPluginId()] ?? [], $submission);
+        $translation['meta'][self::META_FIELD_NAME] = $this->buildData(unserialize($original['meta'][self::META_FIELD_NAME] ?? '') ?: [], $translation[$this->getPluginId()] ?? [], $submission);
```
Comment (PavelLoparev) [inline inc/Smartling/ContentTypes/ExternalContentBeaverBuilder.php:125]: Shouldn't `... ?: []` be `... ?? []`

File: inc/Smartling/Base/SmartlingCoreUploadTrait.php [hunk 2]
```diff
@@ -149,7 +150,7 @@ public function getXMLFiltered(SubmissionEntity $submission): string
             }
 
             $this->prepareFieldProcessorValues($submission);
-            return $this->xmlHelper->xmlEncode($filteredValues, $submission, $source);
+            return $this->xmlHelper->xmlEncode($filteredValues, $submission, array_merge($source, $originalContent));
```
Comment (PavelLoparev) [inline inc/Smartling/Base/SmartlingCoreUploadTrait.php:153]: Could you please show old and new xml file? It's hard to imagine what's inside `original` and `source` content

**Review by PavelLoparev — COMMENTED** [review]
Do we need to add/extend tests for `SmartlingCoreUploadTrait::getXMLFiltered()` and `ExternalContentBeaverBuilder:: applyTranslationValues()`?

**Comment by vsolovei-smartling** [comment]
> Do we need to add/extend tests for `SmartlingCoreUploadTrait::getXMLFiltered()` and `ExternalContentBeaverBuilder:: applyTranslationValues()`?

I don't think so, maybe as part of adding tests for external plugins

File: inc/Smartling/ContentTypes/ExternalContentBeaverBuilder.php (hunk 1 above)
Comment (vsolovei-smartling) [reply inc/Smartling/ContentTypes/ExternalContentBeaverBuilder.php:125]: No, intent was if unserialize results in `false`, use empty array

**Review by PavelLoparev — APPROVED** [review]

**Review by dimitrystd — APPROVED** [review]

**Pull request merged**